File: android/app/src/main/java/com/dismoi/scout/Floating/Layout/Layout.java
package com.dismoi.scout.Floating.Layout;

import android.content.Context;
import android.util.AttributeSet;
import android.view.WindowManager;
import android.widget.FrameLayout;

public class Layout extends FrameLayout {
  private WindowManager windowManager;
  private WindowManager.LayoutParams params;
  private Coordinator layoutCoordinator;

  public void setLayoutCoordinator(Coordinator layoutCoordinator) {
    this.layoutCoordinator = layoutCoordinator;
  }

  Coordinator getLayoutCoordinator() {
    return layoutCoordinator;
  }

  public void setWindowManager(WindowManager windowManager) {
    this.windowManager = windowManager;
  }

  public void create(WindowManager getWindowManager, WindowManager.LayoutParams layoutParams, Coordinator layoutCoordinator) {
    setWindowManager(getWindowManager);
    setViewParams(layoutParams);
    setLayoutCoordinator(layoutCoordinator);
  }

  WindowManager getWindowManager() {
    return this.windowManager;
  }

  public void setViewParams(WindowManager.LayoutParams params) {
    this.params = params;
  }

  public WindowManager.LayoutParams getViewParams() {
    return this.params;
  }

  public Layout(Context context) {
    super(context);
  }

  public Layout(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  public Layout(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
  }
}

File: android/app/src/main/java/com/dismoi/scout/floating/layout/Layout.kt
package com.dismoi.scout.floating.layout

import android.content.Context
import android.util.AttributeSet
import android.view.WindowManager
import android.widget.FrameLayout

open class Layout : FrameLayout {
  var windowManager: WindowManager? = null
  var viewParams: WindowManager.LayoutParams? = null
  var layoutCoordinator: Coordinator? = null
  
  fun create(
    getWindowManager: WindowManager?,
    layoutParams: WindowManager.LayoutParams?,
    layoutCoordinator: Coordinator?
  ) {
    windowManager = getWindowManager
    viewParams = layoutParams
    this.layoutCoordinator = layoutCoordinator
  }

  constructor(context: Context?) : super(context!!) {}
  constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {}
  constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
    context!!, attrs, defStyleAttr
  ) {}
}
